rlp/sedes/lists.py

```diff
@@ -66,7 +66,7 @@ def serialize(self, obj):
         if not is_sequence(obj):
             raise ListSerializationError('Can only serialize sequences', obj)
         if self.strict:
-            if len(self) != len(obj) or len(self) < len(obj):
+            if len(self) != len(obj):
                 raise ListSerializationError('List has wrong length', obj)
```

Comment:
Up for making this error message a little clearer, since mucking around in here?

```py
raise ListSerializationError(
  'List has wrong length %d, expected length %d' % (len(self), len(obj)),
  obj,
)
```